Java/Prims_algorithm_mst.java

```diff
@@ -0,0 +1,86 @@
+import java.util.*; 
+
+class Node 
+{
+    private int v;
+    private int weight;
+    
+    Node(int _v, int _w) { v = _v; weight = _w; }
+    
+    Node() {}
+    
+    int getV() { return v; }
+    int getWeight() { return weight; }
+}
+
+class Main
+{
+    void primsAlgo(ArrayList<ArrayList<Node>> adj, int N)
+    {
+        int key[] = new int[N];
+        int parent[] = new int[N]; 
+        boolean mstSet[] = new boolean[N]; 
+        for(int i = 0;i<N;i++) {
+        	key[i] = 100000000; 
+        	mstSet[i] = false; 
+        }
+
+        key[0] = 0;
+        parent[0] = -1; 
+        
+
+        for(int i = 0;i<N-1;i++) {
+        	int mini = 100000000, u = 0; 
```

Comment:
how can you initailise minimum number as 100000000? what if the smallest number in the test array is greater than your initialised array. Then what will you do?

Reply:
Than this algorithm will not work, since under given constraints of the problem which are

2 ≤ V ≤ 1000
V-1 ≤ E ≤ (V*(V-1))/2
1 ≤ w ≤ 1000

weight cannot exced more than 1000, so under these constraints code works fine.